pkg/clusterconditions/promql/promql.go

```diff
@@ -36,23 +35,26 @@ type PromQL struct {
 	QueryTimeout time.Duration
 }
 
-var promql = &cache.Cache{
-	Condition: &PromQL{
-		HTTPClientConfig: config.HTTPClientConfig{
-			Authorization: &config.Authorization{
-				Type:            "Bearer",
-				CredentialsFile: "/var/run/secrets/kubernetes.io/serviceaccount/token",
-			},
-			TLSConfig: config.TLSConfig{
-				CAFile:     "/etc/tls/service-ca/service-ca.crt",
-				ServerName: "thanos-querier.openshift-monitoring.svc.cluster.local",
+func NewPromQL(kubeClient kubernetes.Interface) *cache.Cache {
+	return &cache.Cache{
+		Condition: &PromQL{
+			kubeClient: kubeClient,
```

Comment:
What's this addition about?

Reply:
Ah, that's from 2de7afff20250f1eae026df8011c954fa096047a .  Ideally the commits would be distinct pivots, with the `kubeClient` stuff all collected together in one, but :shrug:, not a big deal either way.

Reply:
commit history fixed.